Testscripts/Linux/utils.sh

```diff
@@ -118,10 +118,7 @@ function UtilsInit() {
 	if [ -f "$__LIS_CONSTANTS_FILE" ]; then
 		. "$__LIS_CONSTANTS_FILE"
 	else
-		LogErr "Constants file $__LIS_CONSTANTS_FILE missing or not a regular file. Cannot source it!"
-		SetTestStateAborted
-		UpdateSummary "Error: constants file $__LIS_CONSTANTS_FILE missing or not a regular file. Cannot source it!"
-		return 1
+		LogMsg "Constants file $__LIS_CONSTANTS_FILE missing or not a regular file. Cannot source it!"
```

Comment:
What situation could hit this? If we don't source constants.sh, could the test script run correctly?

Reply:
if testparams is empty, the constants.sh will be empty, so sourcing it or not will not impact our testing execution.  On the other side,  if the utils.sh is invoke by other Providers, Controllers, it doesn't make sense to leave a interim Abort state. which may cause a potential bug to provider/controller.

Reply:
I think I understand your intention here @harz566 But when you check out the test result/logging mechanism here, constants.sh file is generated either way with the system generated first line inside the file. So if-statement always hit the true block. Either you change the condition in the if-condition or omit the false block. False block is dead one.

Reply:
no, if we run some bash script (which depends on common methods from utils.sh) in provider/controller body scope, that's not a test case scope (haven't started test case execution, or after test case execution and home folder has been cleaned up),   constants.sh will not be generated at that time or have been removed.   generating an interim Abort state doesn't make sense, and it's a bug hole.    In other words, constants.sh is not always necessary for utils.sh,  so just log a warning message there, is enough.

Reply:
Maybe I misunderstand the logic, but UtilsInit is sourcing in the beginning of the bash script execution in the most of scripts. At that time, constants.sh will be there, when it hit this line. Can you share the TestExecution.log from your case?